src/number_theory/class_number/finite.lean

```diff
@@ -0,0 +1,408 @@
+/-
+Copyright (c) 2021 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen
+-/
+
+import linear_algebra.free_module_pid
+import linear_algebra.matrix.absolute_value
+import number_theory.class_number.admissible_absolute_value
+import number_theory.function_field
+import number_theory.number_field
+import ring_theory.class_group
+import ring_theory.norm
+
+/-!
+# Class numbers of global fields
+In this file, we use the notion of "admissible absolute value" to prove
+finiteness of the class group for number fields and function fields,
+and define `class_number` as the order of this group.
+## Main definitions
+ - `class_group.fintype_of_admissible`: if `R` has an admissible absolute value,
+   its integral closure has a finite class group
+ - `number_field.class_number`: the class number of a number field is the (finite)
+   cardinality of the class group of its ring of integers
+ - `function_field.class_number`: the class number of a number field is the (finite)
+   cardinality of the class group of its ring of integers
```

Comment:
These definitions seem to be missing from the PR.

Reply:
I'll add the lines back in the follow-up PRs.
```suggestion
```